src/app/static/src/app/store/baseline/actions.ts

```diff
@@ -112,15 +112,19 @@ export const actions: ActionTree<BaselineState, DataExplorationState> & Baseline
                 .then((response) => {
                     if (response) {
                         const metadata = response.data;
-                        const pjnz = findResource(metadata, schemas.pjnz);
-                        const pop = findResource(metadata, schemas.population);
-                        const shape = findResource(metadata, schemas.shape);
-                        const survey = findResource(metadata, schemas.survey);
-                        const program = findResource(metadata, schemas.programme);
-                        const anc = findResource(metadata, schemas.anc);
+                        const avaliableResources = rootGetters ? rootGetters["baseline/selectedDatasetAvailableResources"] : {}
+                        const exceptions = { // where DatasetResource keys do not match ADRSchemas keys
+                            pop: "population",
+                            program: "programme"
+                        }
+                        const resources: { [k in keyof DatasetResourceSet]?: DatasetResource | null } = {}
+                        Object.entries(avaliableResources).forEach(([key, value]) => {
+                            const schemaKey = key in exceptions ? exceptions[key as "pop" || "program"] : key
+                            resources[key as keyof typeof resources] = value ? findResource(metadata, schemas[schemaKey as keyof ADRSchemas]) : null
```

Comment:
I think you should revisit this code, I find it very hard to understand. You might want to have a separate method for separate computations.